src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -197,9 +197,19 @@ def base64_encoded_certificate_type(string):
     return cert_data
 
 def datetime_type(string):
-    ''' Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M:%S\'Z\''
-        or  '%Y-%m-%d\'T\'%H:%M\'Z\''. '''
-    date_format = '%Y-%m-%dT%H:%M:%SZ' if len(string) is 20 else '%Y-%m-%dT%H:%MZ'
+    ''' Validates UTC datettime in format '%Y-%m-%d\'T\'%H:%M:%S\'Z\'' or
+    '%Y-%m-%d\'T\'%H:%M\'Z\'' or %Y-%m-%d\'T\'%H:%M\'Z\'' but pads it into
+    the first format. '''
+    try:
+        date_format = '%Y-%m-%dT%H:%M:%SZ'
+        return datetime.strptime(string, date_format)
+    except ValueError:
+        try:
+            padded_string = string[:-1] + ":00Z"
+            return datetime.strptime(padded_string, date_format)
+        except ValueError:
+            padded_string = string[:-1] + ":00:00Z"
+            return datetime.strptime(padded_string, date_format)   
```

Comment:
This is better. However, there is room to improve.

Look at the second try-except structure, it further indents from the first try-except. This is just two exception, what if more? Deeply embedded try-except structures will cause readability issue.

There isn't a universal solution, but you can find a better approach case by case. 

In this case, you need to remember the `return` statement means the function terminates. That implies you don't have to put the second try-except in the first one.

Give it try.